### DISCUSSION TITLE
Field is not used

### DISCUSSION THREAD

**Discussion by timothyarmes** [0no-co/GraphQLSP#222] [Q&A] [answered]
This picture really tells the story:

<img width="649" alt="image" src="https://github.com/0no-co/gql.tada/assets/1092550/03967259-b4eb-4d5b-8cf9-8285dd1b8dda">

VSCode is underlining the fields in the query, telling me that they're not used

<img width="450" alt="image" src="https://github.com/0no-co/gql.tada/assets/1092550/5b36529b-80fc-4d1b-958d-7f6940600a2a">

How can I stop that? I'm not even sure what the cause is.

**Comment by JoviDeCroock** [comment, marked as answer]
This issue will be transferred to https://github.com/0no-co/GraphQLSP where it belongs but you can do `trackFieldUsage: false` in your `tsconfig`. This way of using the documents is one that I have personally not seen before 😅  

So something like

```json
{
  "compilerOptions": {
    "plugins": [
      {
        "name": "@0no-co/graphqlsp",
        "schema": "./schema.graphql",
        "trackFieldUsage": false
      }
    ]
  }
}
```

**Reply by timothyarmes** [reply]
Thanks.  Although it seems to me that you might want to track in some cases and not in others

**Reply by JoviDeCroock** [reply]
I mean, this feature is mainly meant for the front-end case where the reference - document is easier to derive. Where we do things like `client.query()`, I can't really see what you're doing from these limited screenshots but feel free to look at supporting your case.

The code is located [here](https://github.com/0no-co/GraphQLSP/blob/main/packages/graphqlsp/src/fieldUsage.ts#L357-L368) might be as your case is chained function-expressions that it's just adding a hop in the AST traverse. The reference is however part of a function parameter and is then returned and stored into the top-level one which feels a bit hard to track  as you are exporting it and we don't track usage across files as  that is a bit against the grain of [fragment co-location](https://gql-tada.0no.co/guides/fragment-colocation/)
